src/gov/nist/core/LexerCore.java

```diff
@@ -736,7 +736,8 @@ public String number() throws ParseException {
 
         int startIdx = ptr;
         try {
-            if (!isDigit(lookAhead(0))) {
+            char c = lookAhead(0);
+            if (c != '-' && !isDigit(c)) {
```

Comment:
what do you think about my claim? "Even the original code passes, throwing the parseexception (at different part of the logic)."